### PR TITLE
fix: now pot applied to WebEmbedded client too

### PR DESCRIPTION
**Pull request by KagChi**
Enjoy the fix

### CODE REVIEW

File: common/src/main/java/dev/lavalink/youtube/clients/skeleton/NonMusicClient.java [hunk 1]
```diff
@@ -75,8 +75,7 @@ protected JsonBrowser loadTrackInfoFromInnertube(@NotNull YoutubeAudioSourceMana
 
         ClientConfig config = getBaseClientConfig(httpInterface);
 
-        if (status == null) {
-            // Only add embed info if the status is not NON_EMBEDDABLE.
+        if (config.getName() == "WEB_EMBEDDED_PLAYER" || status == null) {
```
Comment (devoxin) [inline common/src/main/java/dev/lavalink/youtube/clients/skeleton/NonMusicClient.java:78]: This isn't how string equality checking works in Java.

**Review by devoxin — CHANGES_REQUESTED** [review]

**Comment by devoxin** [comment]
Please state your reasoning for this PR.

**Comment by KagChi** [comment]
> Please state your reasoning for this PR and what this aims to fix.

YouTube seems to be block poToken workaround on Web client. passing it to WebEmbedded fixes the issue.

**Comment by devoxin** [comment]
Block how? Do requests fail outright with `pot` specified compared to not? Or does it now make no difference whether `pot` is supplied?

To avoid a breaking change I would at least keep the method in the `Web` client. Either no-op it if it breaks any and all requests with it supplied, or keep it as-is if it makes no difference regardless.

**Comment by devoxin** [comment]
Also perhaps embedded clients should expose a flag determining whether they're embedded, so that the appropriate embed client configuration can be applied. Something like
```java
if (status == null || client.isEmbedded()) {
  config.withClientField("clientScreen", "EMBED")
    .withThirdPartyEmbedUrl("https://google.com");
}
```

**Comment by Nansess** [comment]
> Block how? Do requests fail outright with `pot` specified compared to not? Or does it now make no difference whether `pot` is supplied?
> 
> To avoid a breaking change I would at least keep the method in the `Web` client. Either no-op it if it breaks any and all requests with it supplied, or keep it as-is if it makes no difference regardless.

certain flagged IPs when used with PoTokens will result in "please sign in" this seems to be only for the WEB client and doesn't effect WebEmbed

**Comment by KagChi** [comment]
Any updates on this @devoxin ?

**Review by devoxin — APPROVED** [review]

**Pull request merged**